src/spn/algorithms/Gradient.py

```diff
@@ -61,3 +67,38 @@ def prod_gradient_backward(node, parent_result, gradient_result=None, lls_per_no
     assert not np.any(np.isnan(messages_to_children)), "Nans found in iteration"
 
     return messages_to_children
+
+
+def add_node_gradient(node_type, lambda_func):
+    _node_gradient[node_type] = lambda_func
+
+
+def feature_gradient(node, data, node_gradient_functions=_node_gradient, gradient_result=None, lls_per_node=None):
```

Comment:
should we remove gradient_result? or write to it? I believe it's not being used atm. 

can you add to the documentation that we are computing \partial P(X) / \partial X_i?
I believe that's what we are computing, right?

Reply:
Jep, will update that. I believe I wanted to make it possible to write to gradient result if the lls where already calculated... I can remove that, it does not add a significant speedup